mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -192,41 +293,159 @@ impl TendermintCoin {
 
         Ok(TendermintCoin(Arc::new(TendermintCoinImpl {
             ticker,
-            rpc_client,
+            rpc_clients,
             account_id,
             account_prefix: protocol_info.account_prefix,
             priv_key: priv_key.to_vec(),
             decimals: protocol_info.decimals,
             denom,
             chain_id,
+            gas_price: protocol_info.gas_price,
+            avg_block_time,
             sequence_lock: AsyncMutex::new(()),
+            tokens_info: Mutex::new(HashMap::new()),
         })))
     }
 
-    async fn my_account_info(&self) -> MmResult<BaseAccount, TendermintCoinRpcError> {
-        let path = AbciPath::from_str("/cosmos.auth.v1beta1.Query/Account").expect("valid path");
+    // TODO
+    // Save one working client to the coin context, only try others once it doesn't
```

Comment:
In UTXO coin @shamardy implemented the rotation of the electrums:
https://github.com/KomodoPlatform/atomicDEX-API/blob/fb48668f4dc6b9645661551e1be683f93550e53d/mm2src/coins/utxo/rpc_clients.rs#L1641-L1645
Could it be applicable for Tendermint coin?

Reply:
> In UTXO coin @shamardy implemented the rotation of the electrums:
> 
> https://github.com/KomodoPlatform/atomicDEX-API/blob/fb48668f4dc6b9645661551e1be683f93550e53d/mm2src/coins/utxo/rpc_clients.rs#L1641-L1645
> 
> 
> Could it be applicable for Tendermint coin?

Surely it can be. But instead of duplicating codes I guess we can make a generic traits for all platform rpcs.

Btw I haven't checked how exactly it works but I see it locks the connections which means multiple mm2 requests will block each other. Am I missing something?

Reply:
It's also locked at beginning of the function here https://github.com/KomodoPlatform/atomicDEX-API/blob/fb48668f4dc6b9645661551e1be683f93550e53d/mm2src/coins/utxo/rpc_clients.rs#L1549

Reply:
You're right! This shouldn't have much of the performance impact since the mutex guard is dropped before the futures are executed. Anyway we probably need to refactor it! At least refactor the nested mutexes, or even use lock-free data structures.